packages/client/src/api/request/utilits.ts

```diff
@@ -0,0 +1,5 @@
+import {TErrorAPI} from '../typingAPI';
+
+export const isErrorAPI = (data: unknown | TErrorAPI): data is TErrorAPI => {
+	return (data as TErrorAPI).reason !== undefined;
```

Comment:
Если вдруг `data` будет `null` или `undefined` то такая конструкция упадет.  Можно дополнительно добавить проверку:
```suggestion
	return data && (data as TErrorAPI).reason !== undefined;
```